prevent-broker-scale-down-if-contains-partition-replicas.md

```diff
@@ -0,0 +1,58 @@
+
+# Prevent scale down of the brokers if they contain partition replicas
+
+This proposal is about adding the mechanism for preventing the scale down of the brokers in the cluster if one or more partition replicas are hosted on it.
+In the future also, when we introduce the automatic re-balancing when scaling up/down the Kafka Cluster ([proposal](https://github.com/strimzi/proposals/pull/57)), this mechanism would helpful to prevent any data loss when we are scaling down the brokers when they have some partition replicas assigned to them.
+
+## Current situation
+
+Currently, when removing brokers from the Kafka cluster, there is no check to see if these brokers contain any partition replicas.
+The [documentation](https://strimzi.io/docs/operators/latest/configuring.html#scaling-clusters-str) recommends that all topic-partition replicas are reassigned before doing this scaling action, which means that if someone tries to scale down without re-assigning the brokers, it can cause availability issues or data loss.
+
+## Motivation
+
+We should introduce the logic which can detect if the broker which is going to be removed still contains the partition replicas or not.
+If any partition replicas are still assigned to the broker which is going to be removed from the cluster, then we should get some warning in the status of the Kafka resource which will prompt users to do the reassignment and prevent the broker from being removed until the partition replicas are reassigned.
+
+## Proposal
+
+This proposal suggest how we can add the check to detect if the broker still contains any partition replicas and what to do if the broker scale down is not possible.
+
+## Implementation
+
+### Class
+
+A new utility class named `PreventBrokerScaleDownUtils` will be added to the assembly package of cluster operator module
+
+The reason for having this class is to keep all the utilities for checking the broker partition replicas separate such that we don't create a mess in the `KafkaReconciler` class by adding multiple large methods.
+Also, it will help us to test the changes easily.
+
+This class will have utility methods like `canScaleDownBrokers`, `canScaleDownBrokerCheck` and `brokerHasAnyReplicas` methods to check the if broker contains any partition replicas or not.
+
+This class will also contain methods like `topicNames`, `describeTopics` to query the topic metadata.
```

Comment:
As a rule of thumb, it's typically not necessary in a Proposal to go into great detail about the implementation (which will have its own PR). It's not universally true (e.g. adding or removing dependencies), but in this case I think you could drop this section, personally.